### ISSUE TITLE
Feature to suppress specific message of a check

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2804]
http://checkstyle.sourceforge.net/config_filters.html#SuppressionFilter

I would like an option in the "suppression.xml" to suppress specific messages of a check.

Currently I am trying to extend my javadoc collection. I know alot of them don't exist, but I still want to validate the ones that do exist. Most of the javadoc checks, like JavadocTypeCheck, report multiple errors including that the javadoc is missing. To prevent my console from being flooded with missing errors, I would like to suppress them, however "suppressions.xml" only supports suppressing the check as a whole and not the individual errors it reports.
The only option currently is to suppress the check as a whole and hide errors with existing javadocs, which I don't think is a good idea. Or show all errors and try to weed out the 'missing' errors manually, which can be a pain when there are alot.

My example suppressing javadoc is missing message from all checks:

```
<suppressions>
    <suppress messages="Missing a Javadoc comment\." />
</suppressions>
```

With this suppression and my config containing "JavadocTypeCheck", I expect to see no errors reporting my javadoc is missing (javadoc.missing), and expect to see other errors like a tag is missing (type.missingTag).
##

**Comment by romani** [comment]
update need to be done for XML structure http://checkstyle.sourceforge.net/config.html#XML_Structure "Suppressions XML Document" .

`messages` should be RegExp type.

This is issue is blocked by #1571

**Comment by rnveach** [comment]
IT/Test javadoc suppressions ( https://github.com/checkstyle/checkstyle/blob/ee33ed57cd2c88888e9821a2047f587f07283d98/config/suppressions.xml#L81 ) could be changed with this feature to only ignore missing javadocs.

Test currently has 86 javadocs that report errors. 9 of them are just missing authors.
IT currently has 6 javadocs that report errors.

**Comment by timurt** [comment]
I am on it

**Comment by rnveach** [comment]
@timurt Please read full issue, this is blocked by another issue which I am not sure you can do.

**Comment by timurt** [comment]
@rnveach ok, I will look for another issue, thanks

**Comment by rnveach** [comment]
@MEZk @romani Just want to confirm, this is now part of GSoC? I thought GSoC was only going to do XPath suppression?
It was added to the project but not give the label.

**Comment by romani** [comment]
It is not related to GSoC project, could be done at any time .

**Comment by romani** [comment]
fix is merged.
